torch/_inductor/codegen/subgraph.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from typing import Any, Callable, Dict
+
+import torch
+from torch._inductor import ir
+from torch._inductor.codegen.common import KernelTemplate
+from torch._inductor.ir import Buffer, Layout
+from torch._inductor.runtime.benchmarking import benchmarker
+from torch._inductor.virtualized import V
+
+
+log = logging.getLogger(__name__)
+
+
+class SubgraphChoiceCaller(ir.ChoiceCaller):
+    def __init__(
+        self,
+        name: str,
+        input_nodes: list[Buffer],
+        layout: Layout,
+        description: str,
+        gm: torch.fx.GraphModule,
+        example_inputs: list[Any],
+    ) -> None:
+        super().__init__(name, input_nodes, layout, description)
+        self.gm: torch.fx.GraphModule = gm
+        self.example_inputs: list[Any] = example_inputs
+
+    def __str__(self) -> str:
+        return f"SubgraphCaller({self.name})"
+
+    def benchmark(self, *args: list[Any], out: torch.Tensor):
+        import torch._inductor.config as inductor_config
+        from torch._inductor.compile_fx import compile_fx_inner
+
+        # Call the subgraph function with the inputs and output
+        with V.fake_mode as fake_mode:
+            fake_inputs = []
+
+            for arg in self.example_inputs:
+                if isinstance(arg, torch.Tensor):
+                    fake_inputs.append(fake_mode.from_tensor(arg))
+                else:
+                    fake_inputs.append(arg)
+
+        # Don't bother autotuning on Triton here
+        with inductor_config.patch(
+            max_autotune=False,
+            max_autotune_gemm=False,
+            max_autotune_gemm_backends="ATEN",
+        ):
+            benchmark_gm = compile_fx_inner(self.gm, fake_inputs)
+            context = torch._guards.TracingContext.try_get()
+
+            if context:
+                # Reset output strides with nested compilation
+                context.output_strides = []
+
+        out.copy_(benchmark_gm([*args])[0])
+
+        return benchmarker.benchmark_gpu(lambda: benchmark_gm([*args]))
+
+    def hash_key(self) -> str:
+        return "-".join(
```

Comment:
should we include graph in hash key ? E.g., for split-k parameterized by different k ?